lib/simplecov/branches_per_file.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module SimpleCov
+  # This module is responsible for generating a branch coverage report for certain file that is missed by the tests.
+  # Doing something similar to https://ruby-doc.org/stdlib-2.5.3/libdoc/coverage/rdoc/Coverage.html can throw an error.
+  # Errors can be related to Constants, modules are included/extended inside the file but not yet loaded or even exist.
+  # It even can be a programmer's error.
+  module BranchesPerFile
```

Comment:
Frankly, somehow I agree with @klyonrad, but on the other hand, this mechanism will help us to determine the branches in the missed files, which I think is nice to have. 
 After the @klyonrad comment, I added a rescue to print out the error message that problematic file raises, don't you agree with this approach? it would make these modifications not harmful at all

Reply:
Hey - I'm not as worried about the errors (which we can and are catching), but more what random ruby scripts lying around might do. Imagine the following scenario:

1. I wrote ruby script to that sshs into production and runs an expensive operation there, I was carless and included all the credentials. I just dropped it somewhere in my app folder
2. I run my tests
3. The file naturally isn't covered but gets picked up through track files or what not
4. this file requires the file which causes all the ruby code to execute :scream: 
5. Suddenly production is on fire because of that expensive operation and nobody has any idea why :fire: 

I admit it's a worst case scenario. I'm certainly known to have little ruby scripts lying around that do something. They should never be executed as a side effect of a test suite run and to the best of my understanding this makes that possible. We should avoid that at all costs imo.

Reply:
Not like the ultra worst case scenario that you are painting here but at my previous job we literally had a file that **upon loading** was reading stuff from the database, storing into some kind of module/class variable and suddenly the information was shared between the different tests.

The lesson is: `require` does things 😉